dev/com.ibm.ws.concurrent.mp.1.0/src/com/ibm/ws/concurrent/mp/ContextServiceImpl.java

```diff
@@ -75,6 +91,39 @@ protected void deactivate(ComponentContext context) {
         super.deactivate(context);
     }
 
+    /**
+     * Obtain a ManagedExecutor backed by the Liberty global thread pool, without constraints,
+     * and propagating the same types as this ThreadContext service, clearing those which are
+     * configured to be cleared.
+     * If possible, a cached instance is returned. If it doesn't exist yet, then an instance
+     * is lazily created by this method.
+     *
+     * @return ManagedExecutor instance.
+     */
+    private ManagedExecutorImpl getManagedExecutor() {
+        ManagedExecutorImpl executor = managedExecutorRef.get();
+
+        if (executor == null) {
+            String name = new StringBuilder("ManagedExecutor_-1_-1_").append(ManagedExecutorBuilderImpl.instanceCount.incrementAndGet()).toString();
+
+            ConcurrencyProviderImpl concurrencyProvider = (ConcurrencyProviderImpl) ConcurrencyProvider.instance();
+            PolicyExecutor policyExecutor = concurrencyProvider.policyExecutorProvider.create(name);
+            policyExecutor.maxConcurrency(-1).maxQueueSize(-1);
+            // TODO these policy executor instances, as well as those created via ManagedExecutorBuilder are never shut down
+            // and removed from PolicyExecutorProvider's list. This is a memory leak and needs to be fixed.
+
+            executor = new ManagedExecutorImpl(name, policyExecutor, this, concurrencyProvider.transactionContextProvider.transactionContextProviderRef);
+
+            if (!managedExecutorRef.compareAndSet(null, executor)) {
+                // Another thread updated the reference first. Discard the instance we created and use the other.
+                executor.shutdown();
```

Comment:
Will this result in an IllegalStateException since ManagedExecutorImpl extends ManagedExecutorServiceImpl?